torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -588,6 +655,130 @@ std::shared_ptr<ProcessGroup::Work> ProcessGroupMPI::reduce_scatter(
   throw std::runtime_error("ProcessGroupMPI does not support reduce_scatter");
 }
 
+std::shared_ptr<ProcessGroup::Work> ProcessGroupMPI::alltoall_base(
+    at::Tensor& outputTensor,
+    at::Tensor& inputTensor,
+    std::vector<int64_t>& outputSplitSizes,
+    std::vector<int64_t>& inputSplitSizes,
+    const AllToAllOptions& opts) {
+  checkSingleTensorHelper(inputTensor);
+  checkSingleTensorHelper(outputTensor);
+
+  if (outputSplitSizes.size() == 0 && inputSplitSizes.size() == 0) {
+    // We can use alltoall
+    TORCH_CHECK(
+        outputTensor.numel() == inputTensor.numel() &&
+            outputTensor.type() == inputTensor.type(),
+        "Tensors are not equal in size or data type");
+    TORCH_CHECK(
+        outputTensor.size(0) % size_ == 0,
+        "Tensor's dim 0 does not divide equally across group size");
+
+    std::function<void(std::unique_ptr<WorkEntry>&)> runFunc =
+        [opts, this](std::unique_ptr<WorkEntry>& entry) {
+          auto srcdata = (entry->src)[0];
+          auto dstdata = (entry->dst)[0];
+          c10::DeviceGuard guard(srcdata.device());
+          std::unique_lock<std::mutex> globalLock(pgGlobalMutex_);
+          MPI_CHECK(MPI_Alltoall(
+                srcdata.data_ptr(),
```

Comment:
curious, does clang-format say we should to 6-space indent here?

Reply:
clang-format says this should be 4 spaces.

Reply:
Extremely sorry @mrshenli. I am so dumb that I don't know how to use clang-format. Hopefully, I have done it write. 
Please let me know if these are the correct steps to apply clang-format.

- Stage the changes using `git add`
- Run `git-clang-format`
- Stage the new changes again

Last time I searched on web and added few command to my vimrc file, but looks like I didn't do it right. Sorry. Is there any similar way to format python code as well that i should use?